gallery_dl/extractor/deviantart.py

```diff
@@ -1475,11 +1490,26 @@ def _pagination(self, endpoint, params,
                             "oauth:deviantart' and follow the instructions to "
                             "be able to access them.")
                 # "statusid" cannot be used instead
-                if results and "deviationid" in results[0]:
-                    if self.metadata:
-                        self._metadata(results)
-                    if self.folders:
-                        self._folders(results)
+                if results:
+                    if "deviationid" in results[0]:
+                        if self.metadata:
+                            self._metadata(results)
+                        if self.folders:
+                            self._folders(results)
+                    else:
+                        # attempt to fix "deleted" deviations
+                        for result in results:
+                            for item in result.get("items") or ():
+                                if "deviation" not in item or \
+                                        not item["deviation"]["is_deleted"]:
+                                    continue
+                                patch = self._call(
+                                    "/deviation/" +
+                                    item["deviation"]["deviationid"],
+                                    fatal=False)
```

Comment:
Why not use `self.deviation()`?

Reply:
Right now (at least on this branch) neither `metadata` nor `folders` is supposed to work for shared deviations, and I'd rather not make an exception for "deleted" deviations. This will be fixed by [this branch](https://github.com/ClosedPort22/gallery-dl/blob/da-status-expand/gallery_dl/extractor/deviantart.py), which uses `_call` to bypass `_metadata` and `_folders` to make API calls more efficient.